monitoring/main.tf

```diff
@@ -0,0 +1,316 @@
+
+  health_check {
+    interval            = 30
+    path                = "/graph"
+    timeout             = 5
+    healthy_threshold   = 5
+    unhealthy_threshold = 2
+  }
+}
+
+resource "aws_lb_target_group" "grafana" {
+  name     = "grafana-${var.environment}"
+  port     = "${var.grafana_port}"
+  protocol = "HTTP"
+  vpc_id   = "${var.vpc_id}"
+
+  health_check {
+    interval            = 30
+    path                = "/api/health"
+    timeout             = 5
+    healthy_threshold   = 5
+    unhealthy_threshold = 2
+  }
+}
+
+resource "aws_lb_listener_rule" "grafana" {
+  listener_arn = "${module.alb_listener_monitoring.listener_id}"
+  priority     = 2
+
+  action {
+    type             = "forward"
+    target_group_arn = "${aws_lb_target_group.grafana.arn}"
+  }
+
+  condition {
+    field  = "host-header"
+    values = ["grafana.${var.r53_zone_prefix}${var.r53_zone}"]
+  }
+}
+
+data "aws_vpc" "current" {
+  id = "${var.vpc_id}"
+}
+
+resource "aws_security_group_rule" "inbound" {
+  security_group_id = "${var.ecs_sg}"
+  type              = "ingress"
+  from_port         = "${var.prometheus_port}"
+  to_port           = "${var.prometheus_port}"
+  protocol          = "tcp"
+  cidr_blocks       = ["${data.aws_vpc.current.cidr_block}"]
+}
+
+resource "aws_security_group_rule" "allow_ecs_node_monitor" {
+  type              = "ingress"
+  from_port         = 9100
+  to_port           = 9100
+  protocol          = "tcp"
+  security_group_id = "${var.ecs_sg}"
+  self              = true
+}
+
+resource "aws_security_group_rule" "allow_ecs_node_monitor_out" {
+  type              = "egress"
+  from_port         = 9100
+  to_port           = 9100
+  protocol          = "tcp"
+  security_group_id = "${var.ecs_sg}"
+  self              = true
+}
+
+data "aws_lb" "alb" {
+  arn  = "${var.alb_arn}"
+}
+
+resource "aws_route53_record" "monitoring" {
+  zone_id = "${data.aws_route53_zone.root.zone_id}"
+  name    = "monitoring.${var.r53_zone_prefix}${var.r53_zone}"
+  type    = "A"
+
+  alias {
+    name                   = "${data.aws_lb.alb.dns_name}"
+    zone_id                = "${data.aws_lb.alb.zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_route53_record" "grafana" {
+  zone_id = "${data.aws_route53_zone.root.zone_id}"
+  name    = "grafana.${var.r53_zone_prefix}${var.r53_zone}"
+  type    = "A"
+
+  alias {
+    name                   = "${data.aws_lb.alb.dns_name}"
+    zone_id                = "${data.aws_lb.alb.zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_cloudwatch_log_group" "cwlogs" {
+  name              = "monitoring-${var.environment}"
+  retention_in_days = "14"
+
+  tags {
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_s3_bucket" "monitoring_configs_bucket" {
+  bucket = "monitoring-${var.environment}-${var.project}-state"
+  acl    = "private"
+
+  tags {
+    Name        = "monitoring-${var.environment}-${var.project}-configs"
```

Comment:
The `Name` tag is different from the actual bucket name 😅 
Maybe we can remove the `Name` tag altogether, not sure if it's that useful